src/Mutant/Generator/MutationsGenerator.php

```diff
@@ -137,31 +135,21 @@ private function getMutationsFromFile(SplFileInfo $file, bool $onlyCovered, arra
     {
         $initialStatements = $this->parser->parse($file);
 
-        $traverser = new PriorityNodeTraverser();
         $filePath = $file->getRealPath();
         assert(is_string($filePath));
 
-        $mutationsCollectorVisitor = new MutationsCollectorVisitor(
-            $this->mutators,
+        $traverser = $this->traverserFactory->create(
             $filePath,
-            $initialStatements,
             $this->codeCoverageData,
-            $onlyCovered
+            $this->mutators,
+            $onlyCovered,
+            $initialStatements,
+            $extraNodeVisitors
         );
 
-        $traverser->addVisitor(new NotMutableIgnoreVisitor(), 50);
-        $traverser->addVisitor(new ParentConnectorVisitor(), 40);
-        $traverser->addVisitor(new FullyQualifiedClassNameVisitor(), 30);
-        $traverser->addVisitor(new ReflectionVisitor(), 20);
-        $traverser->addVisitor($mutationsCollectorVisitor, 10);
-
-        foreach ($extraNodeVisitors as $priority => $visitor) {
-            $traverser->addVisitor($visitor, $priority);
-        }
-
         $traverser->traverse($initialStatements);
 
-        return $mutationsCollectorVisitor->getMutations();
+        return $traverser->getMutationsCollectorVisitor()->getMutations();
```

Comment:
What if we cut out the `getMutationsCollectorVisitor` here, and  have a `getMutations` method on the traverser itself?